components/chassis.py

```diff
@@ -35,41 +36,71 @@
 from utilities.position import TeamPoses
 
 
-class SwerveModule:
-    DRIVE_GEAR_RATIO = (14.0 / 50.0) * (25.0 / 19.0) * (15.0 / 45.0)
-    STEER_GEAR_RATIO = (14 / 50) * (10 / 60)
-    WHEEL_CIRCUMFERENCE = 4 * 2.54 / 100 * math.pi
-
-    DRIVE_MOTOR_REV_TO_METRES = WHEEL_CIRCUMFERENCE * DRIVE_GEAR_RATIO
-    STEER_MOTOR_REV_TO_RAD = math.tau * STEER_GEAR_RATIO
+@dataclass
+class SwerveConfig:
+    position: Translation2d
+    drive_id: int
+    drive_ratio: float
+    drive_gains: Slot0Configs  # <- kP, kI, kD, kA, kV, kS? we have no feedforward on our steer at the moment
+    steer_id: int
+    steer_ratio: float
+    steer_gains: Slot0Configs
+    encoder_id: int
+    reverse_drive: bool
+    wheel_circumference: float
 
-    # limit the acceleration of the commanded speeds of the robot to what is actually
-    # achiveable without the wheels slipping. This is done to improve odometry
-    accel_limit = 15  # m/s^2
 
+@dataclass
+class CompBotSwerveConfig(SwerveConfig):  # TODO make one for the "testbot" too :)
```

Comment:
This shouldn't be a subclass. The `SwerveConfig` should simply be instantiated with different values.

Reply:
Oh hold on, I see we've put `drive_id` and `steer_id` in this `SwerveConfig` too. I feel like the common things should be in a separate config, so we don't make multiple copies of them?

Reply:
Just chatted with Lucien, we actually just want `SwerveConfig` to have the things that are going to be always common between all the modules on a given robot. So something like

```py
@dataclass
class SwerveConfig:
    drive_ratio: float
    drive_gains: Slot0Configs
    steer_ratio: float
    steer_gains: Slot0Configs
    reverse_drive: bool
    wheel_circumference: float = 4 * 2.54 / 100 * math.pi
```

We can also inject into the chassis the position of the front left module, and assume the chassis is rectangular.

Reply:
would this mean 2 params/injectables/args in `SwerveModule`?

Reply:
`SwerveModule` can continue to separately take its position, motor IDs and encoder ID.